app/src/main/java/org/fossasia/openevent/app/module/event/create/CreateEventFragment.java

```diff
@@ -65,6 +69,17 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
                 getPresenter().createEvent();
         });
 
+        ArrayAdapter<CharSequence> adapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_spinner_item);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+
+        ArrayList<String> currencyCodesArray = new ArrayList<>(CurrencyUtils.getCurrencyCodesList());
+
+        for (int i = 0; i < currencyCodesArray.size(); i++) {
+            adapter.add(currencyCodesArray.get(i));
+        }
```

Comment:
There are two redundant operations happening here. First a list is returned by CurrencyUtils. Then a new list is as currencyCodesArray. Double the memory double the work. Thirdly, Then it is iterated one by one without an iterator but through an index.

Try `adapter.addAll(CurrencyUtils.getCurrencyCodeList())`

If doesn't work, try

```Java
for (String code : CurrencyUtils.getCurrencyCodesList())
    adapter.add(code);
```

Reply:
Secondly, it'd be better if this is moved to the presenter and added lazily but has low priority. If you want, you may move it